readme.md

```diff
@@ -23,3 +23,262 @@ For the revision task, please refer to: [revision.md](revision.md).
+
+ ![alt text](image-20.png)
+ Integer oveflow with conversion\:
+
+ ![alt text](image-21.png)
+ 
+ ![alt text](image-22.png)
+- **Ways to avoid overflow\:** 
+  - Add condition to check the overflow:
+  
+   ![Addition](image.png)
+   ![Subtraction](image-1.png)
+   ![Multiplication](image-2.png)
+   ![Division](image-3.png)
+   - Bounding check\: 
+  
+    ![alt text](image-23.png)
+  - Use available function of GCC compiler:
+  
+   ![alt text](image-4.png)
+## 5. **Format specifier**
+
+| **Types**          | **Format specifier** |
+| ------------------ | :--------------------: |
+| char               | `%c `                  |
+| unsigned char      | `%c`                   |
+| short              | `%hd `                 |
+| string             | `%s `                  |
+| int                | `%d`                   |
+| unsigned int       |` %u`                   |
+| long               | `%li `                 |
+| unsigned long      | `%lu`                  |
+| long long          | `%lli`                 |
+| unsigned long long | `%llu `                |
+| float              | `%f `                  |
+| double             | `%lld`                 |
+| long double        | `%lf`                  |
+
+- **Extended format specifier\:**
+  - `%m.nf`
+    + n\: the number of digits after the commas.
+    + m\: the amount of space used for the output.
+      - m \< 0\:using m space from the left to the right for the output
+      - m \>\: using m space from the right to the left for the output.
+- **Special specifier format\:**
+  - ` %[^\n]`
+  This specifier allows `scanf` to scan character until it encounters a newline character (`\n`).
+  ![alt text](image-15.png)
+  ![alt text](image-16.png)
+  - **`%.*f`\:**
+    - `.` is used to specify precision.
+    - `*` is a placeholder for an integer argument that specifies the precision.
+    ![alt text](image-7.png)
+- **Printf and Scanf\:**
+  - **Printf\:**  is a standard library function in C used for formatted output. It returns the value in format that is declared after the character `%`.
+    Ex\: `%c` returns to character or number, `%d` returns to integer,...
+    - Escape sequences\:
+      -  `\n` for a newline
+      -  `\t` for a tab
+      -  `\b` for a backspace
+      -  `\` for a backslash character
+      -  `'`  and `"` for single and double quote respectively
+      -  `\r` for moving the cursor to the beginning of the line
+      -  `\uXXXX` or `\u{XXXXX}` for representing Unicode characters by their hexadecimal code points.
+  - **Scanf\:** is a standard library function in C used for formatted input. It reads input from the standard input (usually the keyboard) and stores the results into variables specified by the format string.
+    - **Inputting Strings with Spaces using scanf:**
+
+      To input strings with spaces using scanf, you can use the `%[^\n]` format specifier, which reads characters until a newline character is encountered.
```

Comment:
@spm1102 
`#reply_required`
`#for_information`

`^` is used to stop reading, `*` is used to exclude input in `scanf`

Reply:
@Dark-GreenCat oh thank you for this great information!